packages/core/admin/server/utils/index.d.ts

```diff
@@ -18,4 +18,4 @@ type S = {
   'api-token': typeof apiToken;
 };
 
-export function getService<T extends keyof S>(name: T): ReturnType<S[T]>;
+export function getService<T extends keyof S>(name: T): S[T];
```

Comment:
@alexandrebodin @Convly Can you check this typescript change?

Reply:
The `ReturnType` utility in TypeScript is used to give you the ReturnType of a function.:

```ts
type t = ReturnType<() => string> // t will be of type `string`
```

What we currently have is not a valid TypeScript declaration. Here's a simplified example ([TS Playground](https://www.typescriptlang.org/play?#code/C4TwDgpgBAylC8UDeBYAUFKAnA9gGwgC5koAzHHYgCgEoEA+KAcgCMBDLJqAXwG51MAVwDOELMSRkAlgC8Z1OvEatBcpn3Qa06UJCgBzCMBhiAblIDG0RAB4AKlAgAPYBAB2AE2FQA1hBA4pLD0VG5sALZEUHaKjABKRoJYbnbgEDYwANp2ALr0vEA)):

```ts
type S = {
  role: { foo: () => 'bar' };
  user: { fizz: () => 'buzz'};
};

type getService = <T extends keyof S>(name: T) => ReturnType<S[T]>;
```

If ran with TS, the transpiler will complain with the following error:
```ts
Type 'S[T]' does not satisfy the constraint '(...args: any) => any'.
  Type '{ foo: "bar"; } | { fizz: "buzz"; }' is not assignable to type '(...args: any) => any'.
    Type '{ foo: "bar"; }' is not assignable to type '(...args: any) => any'.
      Type '{ foo: "bar"; }' provides no match for the signature '(...args: any): any'.
```

Instead we want the returned type to be the object associated in `S` with the given key `T`.

I hope this is clear 😂

Reply:
Ahah thank you for the details :p I'm not a fan of Typescript so I don't really know what is best :P I remember we modified it recently so that's why I'm asking alex and js to check, they have a better understanding of it than me :p

Reply:
About the `ReturnType`, it was my understanding that services are now functions that returns the actual services helpers (methods, constants, etc...) and are called & then registered with the strapi instance.

If this is the case, then we need the `ReturnType` to get the actual service type & not a regular function.

However, as I quickly read I believe that the API token service is exposing an object rather than a function (same for the other services in admin/core). In this scenario, we indeed need to remove the `ReturnType`.

We need to keep in mind that we'll probably need to refactor this anyway when the Strapi global will be removed (we'll need to pass it down from the function that creates the service).

Reply:
As this utility is focused on the admin/core service I would prefer that we remove the `ReturnType` utility until the refactor is done. This will help in offering auto-completion (for now) when dealing with `getService` and keep the types in sync with the code.

Reply:
Understood :) Good for me then! Thank you for the explanations